src/main/java/core/basesyntax/strategy/OperationStrategy.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ProductDao;
+import core.basesyntax.strategy.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.impl.PurchaseOperationHandler;
+import core.basesyntax.strategy.impl.ReturnOperationHandler;
+import core.basesyntax.strategy.impl.SupplyOperationHandler;
+import java.util.Map;
+
+public class OperationStrategy {
+    private final ProductDao productDao;
+    private final Map<FruitTransaction.Operation, OperationHandler> operationServiceMap;
+
+    public OperationStrategy(ProductDao productDao) {
+        this.productDao = productDao;
+        this.operationServiceMap = createOperationServiceMap();
+    }
+
+    public OperationHandler getOperation(FruitTransaction transaction) {
+        return operationServiceMap.get(transaction.getOperation());
+    }
+
+    private Map<FruitTransaction.Operation, OperationHandler> createOperationServiceMap() {
+        return Map.of(
+                FruitTransaction.Operation.BALANCE,
+                new BalanceOperationHandler(productDao),
+                FruitTransaction.Operation.SUPPLY,
+                new SupplyOperationHandler(productDao),
+                FruitTransaction.Operation.RETURN,
+                new ReturnOperationHandler(productDao),
+                FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler(productDao));
+    }
```

Comment:
not fixed. Please do what @rovein advised: move these map initialization to the Main method and pass it through a constructor into this class

Reply:
To be honest, i don`t know how to fix it...